### ISSUE TITLE
model details about paper Machine Comprehension Using Match-LSTM and Answer Pointer

### ISSUE THREAD

**Issue by InnerPeace-Wu** [shuohangwang/SeqMatchSeq#5]
thx for you time, firstly! 

recently, i tried to re-implement matchLSTM and answer pointer to do QA in squad. And for safety, i basically used the exact model you guys proposed, but my result is much worse, [here is my repo](https://github.com/InnerPeace-Wu/reading_comprehension-cs224n). So i wanna ask:
## did you use dropout and where you put it?

* it seems that you did not mention it in your paper, and i am not a torch guy. My result shows that it overfits seriously. 

## any advise?

I trained with **configuration**: lr=0.02, embed_dim=100(it is lower than yours), hidden representation with 100-dim,  add drop out after representation of context , mix of context and question after matchLSTM. btw, I used GRU in matchLSTM instead of lstm. I only got 62% f1 score on the dev set and around 48% em score. 

do you have any idea where could i go wrong? is it mainly the dims of embedding? since it is more 10% percent worse than your result, it is a big problem. thx a lot.

**Comment by shuohangwang** [comment]
Hi,

You could also add dropout on the embedding layer.

Do you initialise the embedding with Glove or Word2Vec?

Usually, I set a smaller learning rate for adamax or adam, like 0.002 or 0.0002, while I'm not sure about the Tensorflow.

Do you mask the attention values because of the padding? As I'm not familiar to Tensorflow, I just roughly go through your code. I'm sorry about it.

Best,
Shuohang

**Comment by InnerPeace-Wu** [comment]
Thx,
I made a typo, the learning rate is 0.002. and i use **Glove** for initialization. Then maybe its the attention mask issue, i did not try that. 
I may try to fix it later, thx a lot.

## PS

could you please elaborate where you put the dropout layers, apart from the embeddings. And I find out you set dropout rate as **0.4**, is that right? cause that's a heavy drop i think. 

Best,

**Comment by shuohangwang** [comment]
Sorry for the late reply. Hope everything goes well.

The dropout actually can be assigned to every layer, in which case I prefer the dropout rate as 0.3.
